### PR TITLE
Fix for IDENTITY-5249

### PR DESCRIPTION
**Pull request by IsuruIRJ**

### CODE REVIEW

**Comment by thariyarox** [comment]
I have reviewed the fix. Please merge.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/builder/UIBasedConfigurationBuilder.java [hunk 1]
```diff
@@ -166,6 +169,20 @@ private void loadFederatedAuthenticators(AuthenticationStep authenticationStep,
...
+                //When loading the federated IDP configuration from default.xml file in service-providers, we need to
+                // retrieve the federated IDP and load
+                if (federatedAuthenticator == null) {
+                    try {
+                        federatedAuthenticator = IdentityProviderManager.getInstance()
+                                .getIdPByName(federatedIDP.getIdentityProviderName(),
+                                        tenantDomain).getDefaultAuthenticatorConfig();
+                    } catch (IdentityProviderManagementException e) {
+                        throw new FrameworkException("Failed to load the default authenticator for IDP : " +
+                                federatedIDP.getIdentityProviderName());
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/builder/UIBasedConfigurationBuilder.java]: Pass 'e' as a parameter, if not the actual error details will be lost

**Comment by pulasthi7** [comment]
@thariyarox , @IsuruIRJ : Can you get the requested changes done?

**Comment by pulasthi7** [comment]
Improved with #834 hence closing

**Pull request closed without merging**